core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -162,7 +162,11 @@ object StoreUtils extends Logging {
       val primaryKey = {
         v match {
           case PRIMARY_KEY => ""
-          case _ => s"$PRIMARY_KEY ($v, $SHADOW_COLUMN_NAME)"
+          //case _ => s"$PRIMARY_KEY ($v, $SHADOW_COLUMN_NAME)"
+          case _ => s"$PRIMARY_KEY ($SHADOW_COLUMN_NAME)" // Removing partition column for
+          // primary key. This has been done as we dont have support for CLOB in primary key in
+          // desrailized form. Once that's fixed we should uncomment the above line and comment
+          // this line
```

Comment:
First of all we probably don't support partition by complex type( BLOB, ArrayType, MapType, StructType, UDT)  at Gem layer, or do we ? Only remaining thing is CLOB, which we map if  StringType is mentioned in column table. There is no equivalent of CLOB at Spark side . 

What about user partitions based on  (StringType, IntegerType) columns ? We would still like to skip adding them to PK.

Agree to 2nd point for VARCHAR having a first class support at Spark layer or atleast a mechanism to put some metadata to indicate it as a VARCHAR.